### PR TITLE
T31171 Tariff timezones

### PR DESCRIPTION
**Pull request by pwithnall**
https://phabricator.endlessm.com/T31171

### CODE REVIEW

File: panels/updates/cc-tariff-editor.c [hunk 1]
```diff
@@ -497,9 +499,11 @@ cc_tariff_editor_init (CcTariffEditor *self)
...
   gtk_widget_init_template (GTK_WIDGET (self));
 
-  /* Hardcode the default hours (22:00 and 6:00) */
-  self->seconds_to_start = 22 * 60 * 60;
-  self->seconds_to_end = (6 + 24) * 60 * 60;
+  /* Hardcode the default hours (22:00 and 6:00), adding a 24h buffer to ensure
+   * any large negative local timezone offsets don’t shift the UNIX timestamps
+   * below 0. */
+  self->seconds_to_start = 22 * 60 * 60 + (24 * 60 * 60);
+  self->seconds_to_end = (6 + 24) * 60 * 60 + (24 * 60 * 60);
```
Comment (GeorgesStavracas) [inline panels/updates/cc-tariff-editor.c:509]: Nit: perhaps adding an extra `one_day_in_seconds` variable, so that both lines read a bit better

**Review by GeorgesStavracas — APPROVED** [review]
Left one nit, but LGTM. Nice catch, @pwithnall!

File: panels/updates/cc-tariff-editor.c (hunk 1 above)
Comment (GeorgesStavracas) [reply panels/updates/cc-tariff-editor.c:509]: (The variable name is open, I suspect you'll have a better name than that :slightly_smiling_face:)

Comment (pwithnall) [reply panels/updates/cc-tariff-editor.c]: Good suggestion, I have no better name

**Pull request merged**